base/meta.jl

```diff
@@ -223,6 +223,39 @@ function parse(str::AbstractString; raise::Bool=true, depwarn::Bool=true)
     return ex
 end
 
+"""
+    parsefile(filename)
+
+Parse the Julia source file located at `filename` and return a `:toplevel`
+expression that contains all expressions in the file.
+
+julia> expr = Meta.parsefile("hello.jl")
+"""
+function parsefile(filename::AbstractString)
+    fn = String(filename)
+    expr = Base.parse_input_line(read(filename, String); filename = fn)
+    # At this point, `expr` is a `:toplevel` expression that contains
+    # everything in the source file.
+    # `expr` contains multiple `LineNumberNode`s.
+    # Unfortunately, currently some of the `LineNumberNode`s do
+    # not have a filename, i.e. the value of the `file` field is
+    # `nothing`. We go through and replace all of those `nothing`s
+    # with the correct filename. 
+    return _fix_line_number_nodes(expr, Symbol(fn))
```

Comment:
I'd rather not need this. Is there an issue filed about when this happens? Seems like this should be done elsewhere and the fixing deleted from this PR.

Reply:
Not sure if an issue is open. I think @JeffBezanson said that he was looking into it?

Reply:
If you had a reproducible example it might help him find the problem. I haven't noticed this problem myself, so I'm wondering if it only happens in a subset of circumstances. (Or perhaps I'm just blind.)

Reply:
If you run the test case in this PR without fixing the line numbers, then of the three LineNumberNodes, the first one has the correct file name, and the latter two have `nothing` has the file name.

Reply:
Seems to be a bug in the frontend flisp code as indicated by the following:

```
;  _
; |_ _ _ |_ _ |  . _ _
; | (-||||_(_)|__|_)|_)
;-------------------|----------------------------------------------------------

> (jl-parse-all "a = 1\nb=2\n" "blah.jl")
(toplevel (line 1 blah.jl) (= a 1) (line 2)
	  (= b 2))
```